### PR TITLE
Fix unescaping strings for racket, elixir and io

### PR DESCRIPTION
**Pull request by iqbalansari**

### CODE REVIEW

File: io/MalReader.io [hunk 1]
```diff
@@ -28,7 +28,7 @@ MalReader := Object clone do (
     numberRegex := Regex with("^-?[0-9]+$")
 
     read_string := method(token,
-        token exSlice(1, -1) replaceSeq("\\\"", "\"") replaceSeq("\\n", "\n") replaceSeq("\\\\", "\\")
+        token exSlice(1, -1) replaceSeq("\\\\", "\u029e") replaceSeq("\\\"", "\"") replaceSeq("\\n", "\n") replaceSeq("\u029e", "\\")
```
Comment (dubek) [inline io/MalReader.io]: Io doesn't support the `\uNNNN` escape sequence:

```
# io
Io 20110905
Io> ("u029e" == "\u029e")
==> true
Io> "\u029e" size
==> 5
```

Comment (wasamasa) [reply io/MalReader.io]: Chances are you'll have to cook up something similar to https://github.com/kanaka/mal/blob/e27439c7244c32d928000fa1daaeb7e87d4a34b7/gst/util.st#L14

Comment (iqbalansari) [reply io/MalReader.io]: Oh right, what was I thinking! I have pushed a fix. Thanks @wasamasa for the help

**Comment by dubek** [comment]
Looks good.

**Comment by iqbalansari** [comment]
I have pushed a fix for factor too

**Comment by kanaka** [comment]
All tests pass: https://travis-ci.org/kanaka/mal/builds/280849691

I've rebased all the commits together and merged them to master.

**Pull request closed without merging**